### PR TITLE
close transformer state dumps and use separate test output dir in transformer tests

### PR DESCRIPTION
**Pull request by MichaelBelousov**
Haven't been able to reproduce locally by running tests in a loop, but getting CI failures with new transformer resumption tests that are trying to delete for reuse a filepath for state dumps, but the file is locked so sqlite probably still has it open. My current assumption is that closedb is never called when an error is thrown by code editing the db, so this PR just uses `finally` to make sure that errors don't prevent closing the db

### CODE REVIEW

**Review by ColinKerr — APPROVED** [review]

**Comment by aruniverse** [comment]
@mergifyio backport release/3.2.x

File: core/backend/src/test/HubMock.ts [hunk 1]
```diff
@@ -67,17 +67,19 @@ export class HubMock {
...
 
+  protected static get knownTestLocations(): { outputDir: string, assetsDir: string } { return KnownTestLocations; }
+
   /**
    * Begin mocking IModelHub access. After this call, all access to IModelHub will be directed to a [[LocalHub]].
    * @param mockName a unique name (e.g. "MyTest") for this HubMock to disambiguate tests when more than one is simultaneously active.
    * It is used to create a private directory used by the HubMock for a test. That directory is removed when [[shutdown]] is called.
    */
-  public static startup(mockName: LocalDirName) {
+  public static startup(mockName: LocalDirName,) {
```
Comment (pmconne) [inline core/backend/src/test/HubMock.ts:77]: what's with this comma?

Comment (MichaelBelousov) [reply core/backend/src/test/HubMock.ts:77]: removing

**Comment by MichaelBelousov** [comment]
there is one remaining contention but I am testing a fix right now

**Review by pmconne — APPROVED** [review]

File: core/backend/src/test/HubMock.ts (hunk 1 above)
Comment (MichaelBelousov) [reply core/backend/src/test/HubMock.ts]: this should be fixed

**Comment by MichaelBelousov** [comment]
now using a separate directory for the default version0 of LocalHub to prevent a race condition, and also close HubMock in more error cases to prevent false positives about it being left unclosed.

Since this change I have been unable to reproduce the failures in parallel backend/transformer test runs. I am still going to run a few more attempts at reproducing while the CI goes

**Pull request merged**